tools/gen-manifests-diff

```diff
@@ -18,6 +19,16 @@ import tempfile
 arches = ["x86_64", "aarch64", "ppc64le", "s390x"]
 
 
+def find_git_remote_upstream_name():
+    res = subprocess.run(
+        ["git", "remote", "-v"], text=True, capture_output=True, check=True)
+    for line in res.stdout.split("\n"):
+        if m := re.search(r'(\w+)\s+(git@|https:\/\/)github\.com(:|\/)osbuild\/images', line):
+            return m.group(1)
+    # XXX: should we just return an error here?
+    return "uptream"
```

Comment:
Not sure.  If the user has an `upstream` remote, it will work fine.  If they don't, they'll get an error.  The problem is that if they do and it wasn't caught by the regex, then that means they have an `upstream` remote that is not the real upstream.  Perhaps it's fine to use it since that would mean they explicitly decided to have a repo they consider `upstream` that points to a fork or alternative repository.

Reply:
Thanks, yeah, if two of the people using this do not have an "upstream" remote then this is clearly a bad assumption - I made it an error now and if we get this frequently (or even semi-frequently) I think we should implement the idea to just add the remote ourselfs (c.f. https://github.com/osbuild/images/pull/1356#issuecomment-2750997780) (still feeling a bit weary to do this by default because of the changing of user config (even if only temporary)).